libraries/AP_GPS/AP_GPS.h

```diff
@@ -31,8 +31,10 @@
 #ifndef GPS_MAX_RECEIVERS
 #define GPS_MAX_RECEIVERS 2 // maximum number of physical GPS sensors allowed - does not include virtual GPS created by blending receiver data
 #endif
-#ifndef GPS_MAX_INSTANCES
+#if !defined(GPS_MAX_INSTANCES) && GPS_MAX_RECEIVERS > 1
 #define GPS_MAX_INSTANCES  (GPS_MAX_RECEIVERS + 1) // maximum number of GPS instances including the 'virtual' GPS created by blending receiver data
+#else
+#define GPS_MAX_INSTANCES 1
```

Comment:
```suggestion
#if !defined(GPS_MAX_INSTANCES)
#if GPS_MAX_RECEIVERS > 1
#define GPS_MAX_INSTANCES  (GPS_MAX_RECEIVERS + 1) // maximum number of GPS instances including the 'virtual' GPS created by blending receiver data
#endif
```

We probably shouldn't blat the user's pre-defined `MAX_INSTANCES`

Reply:
for the cases with GPS_MAX_RECEIVERS < 2 we do need to override though, don't we? There is only one valid value in that scenario i.e. 1